### ISSUE TITLE
Training models with own data (stero camera) not generating good disparity map results

### ISSUE THREAD

**Issue by xjoshramos** [mrharicot/monodepth#39]
Input is the following image.

![left_14650](https://user-images.githubusercontent.com/4721207/29579025-48acf470-8726-11e7-9c86-9579b45c2ab6.jpg)

Output does not generate disparity map.

![left_14650_disp](https://user-images.githubusercontent.com/4721207/29579065-6b29ad4a-8726-11e7-982d-96d046cf57b6.png)

Here are the parameters used: Along with loss.

parameters(encoder='vgg', height=256, width=512, batch_size=8, num_threads=8, num_epochs=50, do_stereo=False, wrap_mode='border', use_deconv=False, alpha_image_loss=0.85, disp_gradient_loss_weight=0.1, lr_loss_weight=1.0, full_summary=False)
total number of samples: 15875
total number of steps: 99250

batch  98900 | examples/s: 14.65 | loss: 0.49273 | time elapsed: 15.01h | time left: 0.05h
batch  99000 | examples/s: 14.48 | loss: 0.43627 | time elapsed: 15.03h | time left: 0.04h
batch  99100 | examples/s: 14.79 | loss: 0.37097 | time elapsed: 15.04h | time left: 0.02h
batch  99200 | examples/s: 14.49 | loss: 0.20799 | time elapsed: 15.06h | time left: 0.01h

I am not able to get good results on my own data, however I am able to recreate results with kitti data set. And If I input the same image into a model trained on kitti data the result is the following. (disparity map) 

![left_14650_disp](https://user-images.githubusercontent.com/4721207/29579221-fb7eef68-8726-11e7-8459-b0c57f110080.png)

**Comment by xjoshramos** [comment]
The resolution on the new data is JPEG 640x240. I believe this is the problem even though it is fully convolutional?

**Comment by xjoshramos** [comment]
left_test.jpg JPEG 640x240 640x240+0+0 8-bit sRGB 36.1KB 0.000u 0:00.000

**Comment by xjoshramos** [comment]
![left_14650](https://user-images.githubusercontent.com/4721207/29638817-520cb932-880d-11e7-89f4-5c076e82454f.jpg)
![right_14650](https://user-images.githubusercontent.com/4721207/29638821-5525ee18-880d-11e7-9449-9ff5a9091a5a.jpg)

left and right image training data

**Comment by mrharicot** [comment]
Does this image represent a typical training image?
If so your data just seem very hard to train with.
what is the use-case you are trying to train for?
The baseline seems to be very large given the typical distance at which objects are.
In your stereo pair the cable is moving by more than half of the width, you would need to modify the factor in the [get_disp function](https://github.com/mrharicot/monodepth/blob/master/monodepth_model.py#L122).
It also looks like there is significant distortion in your images.
You could try to fine tune one of our models on your data but I think you need a better training set.

**Comment by xjoshramos** [comment]
Thanks for the response.

This is a leap motion stereo image (Near infared)
 
Typically the sensor is used at close ranges. The use case is to generate a disparity map for the given stereo camera. The dataset is not just the cable, but random images collected of the camera pointing at objects indoor and outdoor. 

Also not NIR does not contain texture like RGB. 

Would you say mono_depth works best with object far from the camera? 

What would be considered a better training set? Images similar to kitti? I was able to recreate on the kitti set but not on this data.

**Comment by mrharicot** [comment]
Do you have an example of a more typical image from the dataset you collected?
If the object is really close with respect to the baseline (i.e. will create a large disparity) this model will have trouble dealing with it.

**Comment by poornimajd** [comment]
Hi 
I have the same issue.My training data is from Indian driving dataset.
![0000000_disp](https://user-images.githubusercontent.com/31023599/76184937-c186b700-61f3-11ea-9b8c-72c26770f2a8.jpg)
![0000000](https://user-images.githubusercontent.com/31023599/76184949-ccd9e280-61f3-11ea-91c5-3f827c89c10f.jpg)

Can you please help me out as to why am I getting such results.I am trying to estimate the depth from monocular images obtained from Indian roads.
